contributing/building-a-bento-video-player.md

```diff
@@ -0,0 +1,627 @@
+
+```js
+return <VideoIframe {...props}>
+```
+
+To enable component support for **AMP documents**, our video player element must extend from a base class `VideoBaseElement`. This enables [actions](https://amp.dev/documentation/guides-and-tutorials/learn/amp-actions-and-events/#amp-video-and-other-video-elements_1) and [analytics](https://github.com/ampproject/amphtml/blob/main/extensions/amp-analytics/amp-video-analytics.md), and allows us to define further behavior specific to the AMP layer, like parsing element attributes into Preact props.
+
+This guide covers how to implement video player components that are internally implemented using these Preact and AMP components.
+
+## Getting Started
+
+Start by generating an extension specifying `--bento` and `--nojss`. We name our extension **`amp-fantastic-player`**, ending with `-player` according to our [guidelines for naming a third-party component](../spec/amp-3p-naming.md).
+
+```console
+amp make-extension --bento --nojss --name=amp-fantastic-player
+```
+
+## Directory Structure
+
+A [full directory for a Bento component](./building-a-bento-amp-extension.md#directory-structure) is generated, but this guide will cover the following files in particular:
+
+```
+/extensions/amp-fantastic-player/1.0/
+ ├── base-element.js               # Preact base element
+ ├── component.js                  # Preact implementation
+ ├── amp-my-fantastic-player.js    # Element's implementation
+ └── amp-my-fantastic-player.css   # Custom CSS
+```
+
+## Extend `VideoBaseElement` for AMP
+
+Our `BaseElement` should be a superclass of `VideoBaseElement`. In **`base-element.js`**, we change:
+
+```diff
+  import {MyFantasticPlayer} from './component';
+- import {PreactBaseElement} from '../../../src/preact/base-element';
++ import {VideoBaseElement} from '../../amp-video/1.0/base-element';
+
+- export class BaseElement extends PreactBaseElement {}
++ export class BaseElement extends VideoBaseElement {}
+```
+
+into:
+
+```js
+import {MyFantasticPlayer} from './component';
+import {VideoBaseElement} from '../../amp-video/1.0/base-element';
+
+export class BaseElement extends VideoBaseElement {}
+```
+
+This enables support for AMP actions and analytics, once we map attributes to their prop counterparts in `BaseElement['props']`, and we implement the Preact component.
+
+**Adding this import statement will cause CI to fail**. You should solve this by explicitly allowing the cross-extension import, which is ok for this target file. We add to **`build-system/test-configs/dep-check-config.js`**:
+
+```diff
+{
+  // Extensions can't depend on other extensions.
+  filesMatching: 'extensions/**/*.js',
+  mustNotDependOn: 'extensions/**/*.js',
+  allowlist: [
+    ...
++   // Bento MyFantasticPlayer, amp-my-fantastic-player
++   'extensions/amp-my-fantastic-player/1.0/base-element.js->extensions/amp-video/1.0/base-element.js',
```

Comment:
Might be worth sectioning off the area of dep-check config that currently has YouTube and Vimeo to establish a video-player section. Also, will Bento components need a dep on the `video-manager-impl.js` service or is that only for AMP components?

Reply:
I would prefer that, but I don't know if we should go against the convention in that file.